src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/AdvancedValueComposition.cs

```diff
@@ -0,0 +1,246 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Reflection;
+using System.ComponentModel.Composition.Factories;
+using System.ComponentModel.Composition.Hosting;
+using System.ComponentModel.Composition.UnitTesting;
+using System.ComponentModel.Composition.AttributedModel;
+using System.UnitTesting;
+using Xunit;
+
+namespace System.ComponentModel.Composition
+{
+    public class AdvancedValueComposition
+    {
+        [Fact]
+        public void RepeatedContainerUse()
+        {
+            var container = ContainerFactory.Create();
+            TrivialExporter e = new TrivialExporter();
+
+            CompositionBatch batch = new CompositionBatch();
+            batch.AddPart(e);
+            container.Compose(batch);
+
+            batch = new CompositionBatch();
+            batch.AddPart(new TrivialImporter());
+            container.Compose(batch);
+
+            Assert.True(e.done, "Initialization of importer should have set the done flag on E");
+        }
+
+        [Fact]
+        public void FunctionsFieldsAndProperties()
+        {
+            Consumer c;
+            var container = ContainerFactory.Create();
+
+            CompositionBatch batch = new CompositionBatch();
+            batch.AddPart(new RealAddProvider());
+            batch.AddPart(c = new Consumer());
+            container.Compose(batch);
+
+            Assert.Equal(3, c.op(c.a, c.b)); // "1 + 2 == 3");
+        }
+        [Fact]
+        public void FunctionsFieldsAndProperties2()
+        {
+            Consumer c;
+            var container = ContainerFactory.Create();
+
+            CompositionBatch batch = new CompositionBatch();
+            batch.AddPart(new SubtractProvider());
+            batch.AddPart(c = new Consumer());
+            container.Compose(batch);
+
+            Assert.Equal(-1, c.op(c.a, c.b)); // "1 - 2 == -1");
+        }
+
+        [Fact]
+        public void FunctionsFieldsAndProperties2_StronglyTypedMetadata()
+        {
+            var container = ContainerFactory.CreateWithDefaultAttributedCatalog();
+
+            var exports = container.GetExports<Func<int, int, int>, ITrans_ExportableTest>("Add");
+
+            foreach (var export in exports)
+            {
```

Comment:
Use Assert.All --> this would give us how many items failed. If we do the foreach ourselves it will break the test execution at the first item that failed, instead of testing all.